Messaging.md

```diff
@@ -11,55 +11,89 @@ All data fields are unsigned big-endian unless otherwise specified.
 
 ## Message Format
 
-We reuse the [Lightning Message Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format) and the [Type-Length-Value Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) (TLV).
-To be clear, any encoded binary blob that can be sent over the wire will follow the Lightning Message Format
-while all sub-types internal to these messages will follow the Type-Length-Value Format.
-This means that types on outer-messages will be represented with `u16` integers (defined below) and their length
-is omitted from their encoding because the transport layer has the length in a separate unencrypted field.
-Meanwhile all typed sub-messages (which follow TLV format) will have their types represented using `bigsize` integers
-(defined below) and their lengths (also `bigsize`) are included in their encodings.
+We reuse the [Lightning Message Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format) for messages sent over the wire, the [Type-Length-Value Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) (TLV) for message extensibility, extra features support, as well as for supporting [sub-types](#sub-types) when necessary.
+
+### Wire Messages
+
+Any encoded binary blob that can be sent over the wire will follow the Lightning Message Format.
+This means that wire messages are prefixed with a `u16` field (defined below) and their length is omitted from their encoding because the transport layer has the length in a separate un-encrypted field.
+
+### Type-Length-Value stream
+
+A TLV stream is defined for each wire message, and can be used to extend the protocol and for application specific features.
+See the [Type-Length-Value Format section](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) in the LN BOLTs for specifications.
+
+### Sub-types
+
+Wire messages (and sub-types themselves) contain embedded data structures.
+These sub-types can have three functions:
+* [Plain sub-types](#plain-sub-type) to factor out a number of fields to make specifications clearer,
+* [Sibling sub-types](#sibling-sub-type) to support multiple variants of a field,
+* [Optional sub-types](#optional-sub-type) to support optional fields.
+
+#### Plain sub-type
+
+Plain sub-types do not have any particular format, and their field can simply be replaced in place where they are used.
+
+#### Sibling sub-type
+
+TLV serialization should be used to serialize sibling types.
+Each sibling should be assigned a type identifier.
+The type identifiers are specific to a set of type variants, and can thus be reused across different sub-types.
+Type identifiers are defined within the type definitions, starting from `0` and increasing by `1`.
```

Comment:
I don't think your first point is correct. At the moment the format is not more or less flexible than it would be with this proposal, because we don't expect for anything else than the "TLV" type that is written in the spec.